src/time_axis_scale.rs

```diff
@@ -0,0 +1,67 @@
+use chrono::{DateTime, Duration, Local, NaiveDateTime, Utc};
+/// A TimeAxisScale represents a linear scale for timestamps within a fixed range.
+/// A step in seconds is also expressed and indicates the interval to be used for each tick on the axis.
+///
+/// Time is rendered in the browser's local time.
+use std::{ops::Range, rc::Rc};
+
+use crate::axis::{AxisScale, AxisTick, NormalisedValue};
+
+const DEFAULT_LABEL_FORMAT: &str = "%d-%b";
+
+#[derive(Clone)]
+pub struct TimeAxisScale {
+    time_from: i64,
+    time_to: i64,
+    step: i64,
+    scale: f32,
+    label_format: String,
+}
+
+impl TimeAxisScale {
+    pub fn for_range(
+        range: Range<DateTime<Utc>>,
+        time_step: Duration,
+        label_format: Option<String>,
+    ) -> TimeAxisScale {
+        let time_from = range.start.timestamp();
+        let time_to = range.end.timestamp();
+        let step = time_step.num_seconds();
+        let scale = 1 as f32 / (time_to - time_from) as f32;
+
+        TimeAxisScale {
+            time_from,
+            time_to,
+            step,
+            scale,
+            label_format: label_format.unwrap_or(DEFAULT_LABEL_FORMAT.to_string()),
+        }
+    }
+}
+
+impl AxisScale for TimeAxisScale {
+    fn ticks(&self) -> Rc<Vec<AxisTick>> {
```

Comment:
I think we should return just a Vec at this point... not an Rc. We shouldn't assume the type of box that the user will use.